diff --git a/fbconnect.module b/fbconnect.module
index 82ad43e..b4499ae
--- a/fbconnect.module
+++ b/fbconnect.module
@@ -994,6 +994,7 @@ function theme_fbconnect_login_button($variables) {
     return '<h3>' . $title . '</h3>' . $desc . $link;
   }
   else {
+    $attr['data-scope'] = "email";
     $button = '<fb:login-button ' . drupal_attributes($attr) . '>' . $text . '</fb:login-button>';
     return $button;
   }
